src/PartialCleanDirs.php

```diff
@@ -0,0 +1,248 @@
+<?php
+
+namespace DigipolisGent\Robo\Task\Deploy;
+
+use Robo\Task\BaseTask;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+
+class PartialCleanDirs extends BaseTask
+{
+    /**
+     * Sort by name.
+     *
+     * @const PartialCleanDir::SORT_NAME Sort by name.
+     */
+    const SORT_NAME = 'name';
+
+    /**
+     * This is the time that the file was last accessed, read or written to.
+     *
+     * @const PartialCleanDir::SORT_ACCESS_TIME Sort by access time.
+     */
+    const SORT_ACCESS_TIME = 'access_time';
+
+    /**
+     * This is the last time the actual contents of the file were last modified.
+     *
+     * @const PartialCleanDir::SORT_ACCESS_TIME Sort by modified time.
+     */
+    const SORT_MODIFIED_TIME = 'modified_time';
+
+    /**
+     * This is the time that the inode information was last modified
+     * (permissions, owner, group or other metadata).
+     *
+     * @const PartialCleanDir::SORT_CHANGED_TIME Sort by changed time.
+     */
+    const SORT_CHANGED_TIME = 'changed_time';
+
+    /**
+     * Sorts files and directories by type (directories before files),then by
+     * name.
+     *
+     * @const PartialCleanDir::SORT_TYPE Sort by type.
+     */
+    const SORT_TYPE = 'type';
+
+    /**
+     * The directories to clean.
+     *
+     * @var string
+     */
+    protected $dirs = [];
+
+    /**
+     * The finder used to get the files from the source directories.
+     *
+     * @var \Symfony\Component\Finder\Finder
+     */
+    protected $finder;
+
+    /**
+     * Filesystem component.
+     *
+     * @var \Symfony\Component\Filesystem\Filesystem
+     */
+    protected $fs;
+
+    /**
+     * What the sort the items in the directories by.
+     *
+     * @var string|\Closure
+     */
+    protected $sort;
+
+    /**
+     * Creates a new PartialCleanDir task.
+     *
+     * @param array $dirs
+     *   The directories to clean. Either an array of directories or an array
+     *   keyed by directory with the number of items to keep within this
+     *   directory as value.
+     * @param null|\Symfony\Component\Finder\Finder $finder
+     *   The finder used to get the files from the source folder.
+     * @param null|\Symfony\Component\Filesystem\Filesystem $fs
+     *   Filesystem component to manipulate files.
+     */
+    public function __construct(array $dirs, $finder = null, $fs = null)
```

Comment:
Can't we enforce the type here if the default is null, I thought it was possible: http://php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration